examples/audio_classifier/app/src/main/java/com/google/mediapipe/examples/audioclassifier/fragment/LibraryFragment.kt

```diff
@@ -0,0 +1,343 @@
+        // run on background to avoid block the ui
+        backgroundExecutor?.execute {
+
+            audioClassifierHelper = AudioClassifierHelper(
+                context = requireContext(),
+                currentModel = viewModel.currentModel,
+                classificationThreshold = viewModel.currentThreshold,
+                overlap = viewModel.currentOverlapPosition,
+                numOfResults = viewModel.currentMaxResults,
+                currentDelegate = viewModel.currentDelegate,
+                runningMode = RunningMode.AUDIO_CLIPS,
+                listener = this
+            )
+
+            // prepare media player
+            mediaPlayer = MediaPlayer().apply {
+                setAudioAttributes(
+                    AudioAttributes.Builder()
+                        .setContentType(AudioAttributes.CONTENT_TYPE_MUSIC)
+                        .setUsage(AudioAttributes.USAGE_MEDIA).build()
+                )
+                setDataSource(requireContext(), uri)
+                prepare()
+            }
+
+            val audioFloatArrayData = toFloatArray(targetArray)
+
+            // create audio data to match with model expected length
+            val audioDuration = mediaPlayer!!.duration
+            // calculate the sample rate
+            val expectedSampleRate =
+                audioFloatArrayData.size / (audioDuration / 1000 / AudioClassifierHelper
+                    .EXPECTED_INPUT_LENGTH)
+
+            val resultBundle =
+                audioClassifierHelper?.classifyAudio(
+                    audioFloatArrayData,
+                    expectedSampleRate
+                )
+            resultBundle?.results?.let { audioClassifierResults ->
+                // if the thread is interrupted, skip the next block code.
+                if (Thread.currentThread().isInterrupted) {
+                    return@let
+                }
+                progressExecutor = ScheduledThreadPoolExecutor(1)
+                audioClassifierResults.first().classificationResultList()
+                    ?.get()?.size?.let { maxProgressCount ->
+                        fragmentLibraryBinding.audioProgress.max =
+                            maxProgressCount
+                        val amountToUpdate = audioDuration / maxProgressCount
+                        val runnable = Runnable {
+                            activity?.runOnUiThread {
+                                if (amountToUpdate * fragmentLibraryBinding.audioProgress.progress < audioDuration) {
+                                    var process: Int =
+                                        fragmentLibraryBinding.audioProgress.progress
+                                    val categories =
+                                        audioClassifierResults.first()
+                                            .classificationResultList()
+                                            ?.get()
+                                            ?.get(process)?.classifications()
```

Comment:
Minor nit: can we add the ? to the end of the lines above them so our new lines start with .xxxx()?